R/addin.R

```diff
@@ -69,29 +72,83 @@ confl_create_post_from_Rmd <- function(Rmd_file = NULL, interactive = NULL,
     env = globalenv()
   )
 
+  # set confl setting
   front_matter <- rmarkdown::yaml_front_matter(Rmd_file, "UTF-8")
 
+  front_matter_confl <- front_matter$confl_setting
+
+  confl_setting <- c()
+
+  # set confl setting: base
   if (!is.null(title)) {
-    front_matter$title <- title
+    confl_setting$title <- title
+  } else {
+    confl_setting$title <- front_matter$title
+  }
+
+  if (!is.null(args$tags)) {
+    confl_setting$tags <- args$tags
+  } else {
+    confl_setting <- front_matter_confl$tags
+  }
+
+  if (!is.null(args$space_key)) {
+    confl_setting$space_key <- args$space_key
+  } else {
+    confl_setting$space_key <- front_matter_confl$space_key
   }
 
+  if (!is.null(args$parent_id)) {
+    confl_setting$parent_id <- args$parent_id
+  } else {
+    confl_setting$parent_id <- front_matter_confl$parent_id
+  }
+
+  # set confl_setting for console
+  if (!interactive) {
+    if (!is.null(args$type)) {
+      confl_setting$type <- args$type
+    } else {
+      confl_setting$type <- front_matter_confl$type
+    }
+
+    if (!is.null(args$update)) {
+      confl_setting$update <- args$update
+    } else {
+      confl_setting$update <- front_matter_confl$update
+    }
+
+    if (!is.null(args$use_original_size)) {
+      confl_setting$use_original_size <- args$use_original_size
+    } else {
+      confl_setting$use_original_size <- front_matter_confl$use_original_size
+    }
+  }
+
+
   if (interactive) {
     confl_addin_upload(
       md_file = md_file,
-      title = front_matter$title,
-      tags = front_matter$tags
+      title = confl_setting$title,
+      tags = confl_setting$tags,
+      space_key = confl_setting$space_key,
+      parent_id = confl_setting$parent_id
     )
   } else {
     confl_console_upload(
       md_file = md_file,
-      title = front_matter$title,
-      tags = front_matter$tags,
-      ...
+      title = confl_setting$title,
+      tags = confl_setting$tags,
+      space_key = confl_setting$space_key,
+      type = confl_setting$type,
+      parent_id = confl_setting$parent_id,
+      update = confl_setting$update,
+      use_original_size = confl_setting$use_original_size
```

Comment:
These arguments cannot be `NULL`, which I should have added some check in `confl_console_upload()`...

Reply:
In this point, I will modify to be checked these objects in the confl_create_post_from_Rmd function.